MainMenuSample/Source/Game/ButtonFuctions.cs

```diff
@@ -0,0 +1,306 @@
+{
+    public class ButtonFuctions : Script
+    {
+        public UIControl PlayButton;
+        public UIControl OptionsMainMenuButton;
+        public UIControl QuitButton;
+        public Actor MainMenuText;
+
+        public Actor BlurPanel;
+        public Actor PostProcessingDisabler;
+
+        public string GFX_Preset = "GFX";
+        public string FullscreenPreset = "FULLSCREEN";
+        public string SFX_Preset = "SFX";
+
+        [Tooltip("The dot is placed automatically.")]
+            public string SettingsFileExtension = "FlaxConfig";
+            public string SettingsFileName = "GameSettings";
+
+
+        [Tooltip("Used in the settings file.")]
+            public string SettingsFileTrueTerm = "True";
+        [Tooltip("Used in the settings file.")]
+            public string SettingsFileFalseTerm = "False";
+
+        [Tooltip("Writes a comment on the bottom of the settings file.")]
+            public bool EnableSettingsFileComments;
+
+
+        public UIControl FullscreenButton;
+        public UIControl GraphicsSettingButton;
+        public UIControl AudioButton;
+        public UIControl OptionsBackButton;
+        public Actor OptionsMenuText;
+
+        public SceneReference GameScene;
+
+        //--- Private variables ---\\
+        private bool _audioEnabled;
+        private bool _highGrapicsPreset;
+        private bool _optionsMenuActive;
+        private bool _fullscreen;
+
+        private string _dataPath;
+        private string _directoryPath;
+        private string[] _lastSettingsReadValue;
+
+
+        public override void OnEnable()
+        {
+            PlayButton.Get<Button>().ButtonClicked += PlayButtonFunction;
+            OptionsMainMenuButton.Get<Button>().ButtonClicked += ToggleOptionsMenu;
+            OptionsBackButton.Get<Button>().ButtonClicked += ToggleOptionsMenu;
+            FullscreenButton.Get<Button>().ButtonClicked += ToggleFullscreen;
+            GraphicsSettingButton.Get<Button>().ButtonClicked += ModifyGraphicsSettingQuality;
+            AudioButton.Get<Button>().ButtonClicked += ToggleAudio;
+            QuitButton.Get<Button>().ButtonClicked += QuitButtonFunction;
+
+            // Generates the path to the settings file.
+            string fileName = SettingsFileName + "." + SettingsFileExtension;
+            
+            _dataPath = StringUtils.CombinePaths(Globals.ProductLocalFolder, fileName);
+            Debug.Log("Settings file location: " + _dataPath);
+
+            if (!File.Exists(_dataPath))
+            {
+                if (!Directory.Exists(Globals.ProductLocalFolder))
+                {
+                    Directory.CreateDirectory(Globals.ProductLocalFolder);
+                }
+                WriteSettingsFile(true, false, true);
+            }
+
+            _optionsMenuActive = true;
+            ToggleOptionsMenu(OptionsMainMenuButton.Get<Button>());
+
+            ReadSettingsFile();
+
+            string lastRead_Formed = "";

```

Comment:
I can't really encourage the use of this form of parser. Seems pretty brittle and hack-ish